projects/observability/src/shared/dashboard/widgets/charts/cartesian-widget/interactions/cartesian-explorer-context-menu/cartesian-explorer-context-menu.component.ts

```diff
@@ -0,0 +1,65 @@
+import { ChangeDetectionStrategy, Component, Inject } from '@angular/core';
+import { IconType } from '@hypertrace/assets-library';
+import { TimeRangeService } from '@hypertrace/common';
+import { ButtonStyle, POPOVER_DATA } from '@hypertrace/components';
+import { CartesianSelectedData } from '../../../../../../components/cartesian/chart-interactivty';
+import { CartesainExplorerNavigationService } from '../cartesian-explorer-navigation.service';
+
+@Component({
+  selector: 'ht-cartesian-explorer-context-menu',
+  styleUrls: ['./cartesian-explorer-context-menu.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  template: `
+    <div class="context-menu-container">
+      <div *ngFor="let menu of menus">
+        <div class="context-menu">
+          <ht-button [label]="menu.name" [icon]="menu.icon" [display]="display" (click)="menu.onClick()"></ht-button>
+        </div>
+
+        <ht-divider></ht-divider>
+      </div>
+    </div>
+  `
+})
+export class CartesianExplorerContextMenuComponent<TData> {
+  public menus?: ContextMenu[] = [
+    {
+      name: 'Set Time Range',
+      icon: IconType.Alarm,
+      onClick: () => this.setTimeRangeHandler()
+    },
+    {
+      name: 'Explore',
+      icon: IconType.ArrowUpRight,
+      onClick: () => this.explorerNavigationHandler()
+    }
+  ];
+
+  public display: string = ButtonStyle.PlainText;
+  public selectionData: CartesianSelectedData<TData>;
+
+  public constructor(
+    @Inject(POPOVER_DATA) data: CartesianSelectedData<TData>,
+    private readonly cartesainExplorerNavigationService: CartesainExplorerNavigationService,
+    private readonly timeRangeService: TimeRangeService
+  ) {
+    this.selectionData = data;
+  }
+
+  public readonly explorerNavigationHandler = () => {
+    this.cartesainExplorerNavigationService.navigateToExplorer(
+      this.selectionData.timeRange.startTime,
+      this.selectionData.timeRange.endTime
+    );
+  };
+
+  public readonly setTimeRangeHandler = () => {
+    this.timeRangeService.setFixedRange(this.selectionData.timeRange.startTime, this.selectionData.timeRange.endTime);
+  };
+}
+
+export interface ContextMenu {
```

Comment:
this is a local type, doesn't need to be exported

Reply:
using in test file, ie why exported

Reply:
Commented on the test file too, it shouldn't be using it - it's defining what it's trying to test, rather than just interacting with the component.

If for some reason we did need to use it in another file and thus export it, the naming becomes more important because it shows up in autocomplete. In that case, we'd want to change the name to clarify that it's only used in here (e.g. `CartesianHandlerContextMenuIte`) - but again, we shouldn't need to.